v-next/hardhat/src/internal/builtin-plugins/network-manager/edr/types/hardfork.ts

```diff
@@ -18,21 +22,84 @@ export enum HardforkName {
   CANCUN = "cancun",
 }
 
-const HARDFORK_ORDER = Object.values(HardforkName);
+export enum OpHardforkName {
+  BEDROCK = "bedrock",
+  REGOLITH = "regolith",
+  CANYON = "canyon",
+  ECOTONE = "ecotone",
+  FJORD = "fjord",
+  GRANITE = "granite",
+  HOLOCENE = "holocene",
+}
 
-export const LATEST_HARDFORK: HardforkName =
-  HARDFORK_ORDER[HARDFORK_ORDER.length - 1];
+const L1_HARDFORK_ORDER = Object.values(L1HardforkName);
+const OP_HARDFORK_ORDER = Object.values(OpHardforkName);
+
+export function getHardforks(chainType: ChainType): string[] {
+  return chainType === OPTIMISM_CHAIN_TYPE
+    ? OP_HARDFORK_ORDER
+    : L1_HARDFORK_ORDER;
+}
+
+const L1_LATEST_HARDFORK: L1HardforkName =
+  L1_HARDFORK_ORDER[L1_HARDFORK_ORDER.length - 1];
+const OP_LATEST_HARDFORK: OpHardforkName =
+  OP_HARDFORK_ORDER[OP_HARDFORK_ORDER.length - 1];
+
+export function getLatestHardfork(chainType: ChainType): string {
+  return chainType === OPTIMISM_CHAIN_TYPE
+    ? OP_LATEST_HARDFORK
+    : L1_LATEST_HARDFORK;
+}
```

Comment:
Important to note for EDR:

What EDR refers to as `opLatestHardfork()` or `l1LatestHardfork` refers to the latest activated hardfork. It could however be that internally there are supported pre-release hardfork. E.g. when clients are starting to develop Osaka, EDR might already have that hardfork as an option with a preliminary implementation.

This would adversely affect this logic if Hardhat adds Osaka  to the `HardforkName` enum.

Reply:
I agree this needs another iteration. Maybe we could assign a "state" to each hardfork and pick the latest one that isn’t a pre-release. That said, this might be good enough for the alpha.